packages/Popover/src/Popover.js

```diff
@@ -0,0 +1,517 @@
+            contentRefPos.width / 2
+          );
+      }
+    }
+  }
+
+  calcWithoutPortalLeft({ alignment, justification }) {
+    const contentRefPos = this.getRefPosition(this.contentRef);
+
+    if (contentRefPos && alignment) {
+      switch (alignment) {
+        case 'left':
+          return `${0 - contentRefPos.width}px`;
+
+        case 'right':
+          return '100%';
+      }
+    }
+
+    if (justification) {
+      switch (justification) {
+        case 'left':
+          return 0;
+
+        case 'right':
+          return `calc(100% - ${contentRefPos.width}px)`;
+
+        case 'center-horizontal':
+          return `calc(50% - ${contentRefPos.width}px / 2)`;
+      }
+    }
+  }
+
+  // Get transform styles for position object
+  getTransform(alignment) {
+    const transformAmount = 12;
+    const scaleAmount = 0.8;
+
+    switch (alignment) {
+      case 'top':
+        return `translate3d(0, ${transformAmount}px, 0) scale(${scaleAmount})`;
+
+      case 'bottom':
+        return `translate3d(0, -${transformAmount}px, 0) scale(${scaleAmount})`;
+
+      case 'left':
+        return `translate3d(${transformAmount}px, 0, 0) scale(${scaleAmount})`;
+
+      case 'right':
+        return `translate3d(-${transformAmount}px, 0, 0) scale(${scaleAmount})`;
+    }
+  }
+
+  // Constructs the transform origin for any given pair of alignment / justification
+  getTransformOrigin({ alignment, justification }) {
+    let x, y;
+
+    switch (alignment) {
+      case 'left':
+        x = 'right';
+        break;
+
+      case 'right':
+        x = 'left';
+        break;
+
+      case 'bottom':
+        y = 'bottom';
+        break;
+
+      case 'top':
+        y = 'top';
+        break;
+    }
+
+    switch (justification) {
+      case 'left':
+        x = 'left';
+        break;
+
+      case 'right':
+        x = 'right';
+        break;
+
+      case 'bottom':
+        y = 'top';
+        break;
+
+      case 'top':
+        y = 'bottom';
+        break;
+
+      case 'center-horizontal':
+        x = 'center';
+        break;
+
+      case 'center-vertical':
+        y = 'center';
+        break;
+    }
+
+    return `${x} ${y}`;
+  }
+
+  contentRef = React.createRef();
+  createRefEl = React.createRef();
+
+  render() {
+    const {
+      children,
+      active,
+      className,
+      withoutPortal,
+      refEl,
+      ...rest
+    } = this.props;
+
+    // const { position } = this.state;
+    const position = this.calculatePosition();
+
+    const Root = withoutPortal ? Fragment : Portal;
+
+    const activeStyle = {
+      transform: 'translate3d(0, 0, 0) scale(1)',
+      opacity: 1,
+      pointerEvents: 'initial',
+    };
+
+    const style = { ...position, ...(active && activeStyle) };
+
+    return (
+      <>
+        <div ref={this.createRefEl} className={divRef} />
+        <Root>
+          <div
+            {...rest}
+            ref={this.contentRef}
+            className={cx(className, rootPopoverStyle)}
```

Comment:
To note: with `cx`, order matters, and the last param will take the highest precedence. Let's be sure that `className` always comes last.